pandas/core/indexes/range.py

```diff
@@ -435,6 +435,10 @@ def _shallow_copy(self, values, name: Hashable = no_default):
 
         if values.dtype.kind == "f":
             return Float64Index(values, name=name)
+        unique_diffs = np.unique(np.diff(values))
```

Comment:
libalgos.unique_deltas?

Reply:
Ah cool I'll try this out. 

This is more of an experiment to see what breaks when returning a RangeIndex. Apparently a lot of things